### PR TITLE
Chore/1489 deprecate aws id generator

### PR DESCRIPTION
**Pull request by sreeo**
Fixes #
Design discussion issue (if applicable) #

## Changes
Deprecates the XrayIdGenerator in the opentelemetry core sdk. 
Relates [#1489](https://github.com/open-telemetry/opentelemetry-rust/issues/1489)

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,13 @@
 # Changelog
 
+## Deprecation Notice
```
Comment (lalitb) [inline opentelemetry-sdk/CHANGELOG.md:3]: should this instead be a "Migration Notice" or "Functionality moved"  or something similar? As this component is moved to another crate without an immediate plan to remove it from the ecosystem entirely?

File: opentelemetry-sdk/CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
+
+XrayIdGenerator has been migrated to the opentelemetry-aws crate starting from 0.10.0. [#33](https://github.com/open-telemetry/opentelemetry-rust-contrib/pull/33)
```
Comment (cijothomas) [inline opentelemetry-sdk/CHANGELOG.md:5]: the message is not very clear. Could you rephrase so that it is clear that we are referrung to 0.10.0 of the aws create.

Also, this can be under a "##Removed" section.

Comment (lalitb) [reply opentelemetry-sdk/CHANGELOG.md:5]: I guess has been deprecated in this crate, not removed yet.

**Review by lalitb — APPROVED** [review]

File: opentelemetry-sdk/CHANGELOG.md (hunk 2 above)
Comment (hdost) [reply opentelemetry-sdk/CHANGELOG.md:5]: It can however be moved inside of the version. I think this may have been inspired by my jaeger deprecation, but in this case we're only deprecating a component. Where as for jaeger we're deprecating the crate. 

Could be:

```
# Changelog

## vNext

### Deprecated
...

```

Comment (sreeo) [reply opentelemetry-sdk/CHANGELOG.md]: how about this?
`
### Deprecated

- XrayIdGenerator in the opentelemetry-sdk has been deprecated and moved to version 0.10.0 of the opentelemetry-aws crate.
`
